### DISCUSSION TITLE
Using `createServerSideHelpers` without `appRouter`

### DISCUSSION THREAD

**Discussion by ewnd9** [trpc/trpc#4440] [Q&A] [answered]
Hello, I have 2 apps:

- backend with trpc
- frontend Next.js app for SSR

How should I instantiate `createServerSideHelpers` if I don't have access to `appRouter`? I suppose there should be a way to transform `TRPCProxyClient` to act as router.

Or what are the alternatives? (`ssr:true` works, but I need access to `getServerSideProps`, see https://github.com/trpc/trpc/issues/596)

**Comment by ewnd9** [comment]
I was able to get `rehydrate` working locally by forking `ssg.ts`/`ssgProxy.ts` and replacing [`callProcedure` calls](https://github.com/trpc/trpc/blob/d96d9c2b42090d8e4a91ba507d99225e632e6083/packages/react-query/src/ssg/ssg.ts#L41C16-L47) with `trpcClient.query(...pathAndInput)`

**Reply by EmericW** [reply]
I would like this feature as well and implemented a similar solution to what you describe [here](https://github.com/trpc/trpc/pull/4482)

**Reply by KATT** [reply]
I left some comments on that PR

**Comment by nicholashamilton** [comment]
I have an express backend with tRPC and a frontend in NextJS. 

I'm really in need of a way to use prefetchQuery/ prefetchInfiniteQuery without the appRouter and call dehydrate into props trpcState.

**Comment by EmericW** [comment, marked as answer]
This [PR](https://github.com/trpc/trpc/pull/4547) has been merged and is available in [v10.32.0](https://github.com/trpc/trpc/releases/tag/v10.32.0)

**Reply by nicholashamilton** [reply]
Thank you for working on this!
